lib/Runtime/Language/JavascriptOperators.inl

```diff
@@ -10,46 +10,57 @@ namespace Js
     {
         AssertMsg(aValue != nullptr, "GetTypeId aValue is null");
 
-        if (TaggedInt::Is(aValue))
+        if (RecyclableObject::Is(aValue))
         {
-            return TypeIds_Integer;
+            auto typeId = RecyclableObject::FromVar(aValue)->GetTypeId();
+#if DBG
+            auto isExternal = RecyclableObject::FromVar(aValue)->CanHaveInterceptors();
+            AssertMsg(typeId < TypeIds_Limit || isExternal, "GetTypeId aValue has invalid TypeId");
+#endif
+            return typeId;
         }
 #if FLOATVAR
-        else if (JavascriptNumber::Is_NoTaggedIntCheck(aValue))
+        else if (TaggedInt::Is(aValue))
         {
+            return TypeIds_Integer;
+        }
+        else
+        {
+            Assert(JavascriptNumber::Is_NoTaggedIntCheck(aValue));
             return TypeIds_Number;
         }
-#endif
+#else
         else
         {
-            auto typeId = RecyclableObject::FromVar(aValue)->GetTypeId();
-#if DBG
-            auto isExternal = RecyclableObject::FromVar(aValue)->CanHaveInterceptors();
-            AssertMsg(typeId < TypeIds_Limit || isExternal, "GetTypeId aValue has invalid TypeId");
-#endif
-            return typeId;
+            return TypeIds_Integer;
```

Comment:
>TypeIds_Integer [](start = 19, length = 15)

GetTypeId called with a non-valid var will now return TypeIds_Integer on x86 and TypeIds_Number on x64?

Reply:
Can you please elaborate `with a non-valid var` ?

Reply:
For example. a pointer to a FunctionBody

Reply:
In such case what we would return before?

Reply:
@rajatd are you referring to an undocumented usage? Or a concern?